src/cp/slipharden.cxx

```diff
@@ -1,21 +1,398 @@
+                                         const History & history, 
+                                         Lattice & L, double T, 
+                                         const History & fixed, 
+                                         std::vector<std::string> ext) const
+{
+  return history.subset(ext).derivative<double>().zero();
+}
+
+History SlipHardening::d_hist_d_h_ext(const Symmetric & stress,
+                                      const Orientation & Q,
+                                      const History & history,
+                                      Lattice & L, double T, const SlipRule & R,
+                                      const History & fixed,
+                                      std::vector<std::string> ext) const
+{
+  return blank_hist().history_derivative(history.subset(ext)).zero();
+}
+
 bool SlipHardening::use_nye() const
 {
   return false;
 }
 
+FixedStrengthHardening::FixedStrengthHardening(
+    std::vector<std::shared_ptr<Interpolate>> strengths) :
+      strengths_(strengths)
+{
+  init_cache_();
+}
+
+std::string FixedStrengthHardening::type()
+{
+  return "FixedStrengthHardening";
+}
+
+std::unique_ptr<NEMLObject> FixedStrengthHardening::initialize(ParameterSet & params)
+{
+  return neml::make_unique<FixedStrengthHardening>(
+      params.get_object_parameter_vector<Interpolate>("strengths"));
+}
+
+ParameterSet FixedStrengthHardening::parameters()
+{
+  ParameterSet pset(FixedStrengthHardening::type());
+
+  pset.add_parameter<std::vector<NEMLObject>>("strengths");
+
+  return pset;
+}
+
+std::vector<std::string> FixedStrengthHardening::varnames() const
+{
+  return {};
+}
+
+void FixedStrengthHardening::set_varnames(std::vector<std::string> vars)
+{
+  init_cache_();
+}
+
+void FixedStrengthHardening::populate_history(History & history) const
+{
+}
+
+void FixedStrengthHardening::init_history(History & history) const
+{
+}
+
+double FixedStrengthHardening::hist_to_tau(size_t g, size_t i, 
+                                           const History & history,
+                                           Lattice & L,
+                                           double T, const History & fixed) const
+{
+  return strengths_[L.flat(g,i)]->value(T);
+}
+
+History FixedStrengthHardening::d_hist_to_tau(size_t g, size_t i, 
+                                              const History & history,
+                                              Lattice & L,
+                                              double T, 
+                                              const History & fixed) const
+{
+  return cache(CacheType::DOUBLE);
+}
+
+History FixedStrengthHardening::hist(const Symmetric & stress, 
+                                     const Orientation & Q,
+                                     const History & history, 
+                                     Lattice & L, double T, const SlipRule & R, 
+                                     const History & fixed) const
+{
+  return blank_hist();
+}
```

Comment:
No on the move to header.  Maybe on "final."  We should go through all the code at some point and do this (and see if it makes any difference, it may not).